src/App.tsx

```diff
@@ -1,61 +1,154 @@
+      completed: false,
+      user: todoUser,
+    };
+
+    setTodoList((prevTodos) => [...prevTodos, todoToAdd]);
+
+    nextTodoId += 1;
+
+    return setNewTodo(emptyTodo);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="POST" onSubmit={handleAddTodo}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="todoTitle">Title: </label>
+          <input
+            id="todoTitle"
+            type="text"
+            placeholder="Enter a title"
+            data-cy="titleInput"
+            name="title"
+            value={newTodo.title}
+            onChange={handleChange}
+          />
+          {!newTodo.isTitleValid && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label htmlFor="todoUser">User: </label>
+          <select
+            id="todoUser"
+            data-cy="userSelect"
+            name="userName"
+            value={newTodo.userName}
+            onChange={handleChange}
+          >
+            <option value="" disabled>
+              Choose a user
+            </option>
+            {usersFromServer.map((user) => (
+              <option key={user.id} value={user.name}>
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
+          {!newTodo.isUserNameValid && (
+            <span className="error">Please choose a user</span>
+          )}
         </div>
 
         <button type="submit" data-cy="submitButton">
           Add
         </button>
       </form>
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      {todos.length && <TodoList todos={todoList} />}
```

Comment:
```suggestion
      {!!todos.length && <TodoList todos={todoList} />}
```
Otherwise you will see 0 on the screen when the list of todo is empty, which is unexpected behaviour